src/sentry/mediators/sentry_apps/updater.py

```diff
@@ -49,6 +52,24 @@ def call(self):
         self.sentry_app.save()
         return self.sentry_app
 
+    @if_param("features")
+    def _update_features(self):
+        if not self.user.is_superuser and self.sentry_app.status == SentryAppStatus.PUBLISHED:
+            raise APIError("Cannot update features on a published integration.")
+        # Delete any unused features
+        unused_features = IntegrationFeature.objects.filter(
```

Comment:
I think you can chain `delete()` instead of looping:
```py
unused_features = IntegrationFeature.objects.filter(
    target_id=self.sentry_app.id, 
    target_type=IntegrationTypes.SENTRY_APP.value,
).exclude(feature__in=list(self.features)).delete()
```